src/MagicWormhole/Internal/FileTransfer.hs

```diff
@@ -29,12 +29,26 @@ import System.Posix.Types (FileOffset)
 data Offer
   -- | A simple text message.
   = Message Text
+  -- | Offer a File with filename and size.
   | File FilePath FileOffset
+  -- | Offer a Directory
+  | Directory
+    Text
+    -- ^ Mode. Currently always "zipfile/deflated".
+    Text
+    -- ^ Directory Name.
+    Integer
+    -- ^ size of the transmitted compressed data in bytes
+    Integer
+    -- ^ estimated total size of the uncompressed directory
+    Integer
```

Comment:
Again, lack of detailed knowledge of the protocol is preventing me from being authoritative: are these all really unbounded? Would, e.g., `Int64` be more appropriate?

Reply:
I am not aware of any upper bounds either. After establishing the tcp pipe, we send records. Each of the record itself has a size upper bound of 4GB. But I am not aware of other bounds for the zipfiles representing directories or number of files in the directories etc.